### PR TITLE
MCOL-5500 This patch adds cgroup v2 support with some sanity checks for

### PR DESCRIPTION
**Pull request by drrtuy**
values reported by cgroups v1

### CODE REVIEW

**Review by mariadb-LeonidFedorov — APPROVED** [review]

**Review by mariadb-RomanNavrotskiy — APPROVED** [review]

File: utils/common/cgroupconfigurator.cpp [hunk 1]
```diff
@@ -171,7 +179,7 @@ uint32_t CGroupConfigurator::getNumCores()
       ret = getNumCoresFromProc();
   }
 
-  // cout << "There are " << ret << " cores available" << endl;
+  cout << "There are " << ret << " cores available" << endl;
```
Comment (tntnatbry) [inline utils/common/cgroupconfigurator.cpp:182]: Please comment out this line.

File: utils/common/cgroupconfigurator.cpp [hunk 2]
```diff
@@ -188,11 +196,11 @@ uint64_t CGroupConfigurator::getTotalMemory()
   {
     ret = getTotalMemoryFromCGroup();
 
-    if (ret == 0)
+    if (ret == 0 || ret == std::numeric_limits<uint64_t>::max())
       ret = getTotalMemoryFromProc();
   }
 
-  // cout << "Total mem available is " << ret << endl;
+  cout << "Total mem available is " << ret << endl;
```
Comment (tntnatbry) [inline utils/common/cgroupconfigurator.cpp:203]: Please comment out this line.

File: utils/common/cgroupconfigurator.cpp [hunk 3]
```diff
@@ -227,61 +223,88 @@ uint64_t CGroupConfigurator::getTotalMemoryFromProc()
...
 uint64_t CGroupConfigurator::getFreeMemory()
 {
   uint64_t ret;
-
   if (!cGroupDefined)
     ret = getFreeMemoryFromProc();
   else
   {
     uint64_t usage = getMemUsageFromCGroup();
+    cout << "usage " << usage << endl;
```
Comment (tntnatbry) [inline utils/common/cgroupconfigurator.cpp:280]: Please comment out this line.

File: utils/common/cgroupconfigurator.cpp [hunk 4]
```diff
@@ -141,7 +149,7 @@ uint32_t CGroupConfigurator::getNumCoresFromCGroup()
     first = last + 1;
   }
 
-  // cout << "found " << cpus << " CPUS in the string " << cpusString << endl;
+  cout << "found " << cpus << " CPUS in the string " << cpusString << endl;
```
Comment (tntnatbry) [inline utils/common/cgroupconfigurator.cpp:152]: Please comment out this line.

File: utils/common/cgroupconfigurator.cpp [hunk 5]
```diff
@@ -227,61 +223,88 @@ uint64_t CGroupConfigurator::getTotalMemoryFromProc()
...
+    cout << "usage " << usage << endl;
 
     if (usage == 0)
       ret = getFreeMemoryFromProc();
     else
       ret = getTotalMemory() - usage;
   }
 
-  // cout << "free memory = " << ret << endl;
+  cout << "free memory = " << ret << endl;
```
Comment (tntnatbry) [inline utils/common/cgroupconfigurator.cpp:288]: Please comment out this line.

**Review by tntnatbry — COMMENTED** [review]
@drrtuy I have highlighted 5 print out statements to stdout that should be commented out.

**Pull request merged**